Michelf/MarkdownExtra.php

```diff
@@ -1292,10 +1295,9 @@ protected function doFencedCodeBlocks($text) {
 					(?:~{3,}|`{3,}) # 3 or more tildes/backticks.
 				)
 				[ ]*
+				# #2: class name (optional) and #3: attributes (optional)
 				(?:
-					\.?([-_:a-zA-Z0-9]+) # 2: standalone class name
-				|
-					'.$this->id_class_attr_catch_re.' # 3: Extra attributes
+					\.?([-_:a-zA-Z0-9]+)?[ ]*('.$this->id_class_attr_catch_re.')?
```

Comment:
This regex has a two issues:

1. The optional dot before the language name must not be allowed by itself, it must be followed by a language name.
2. `$this->id_class_attr_catch_re` already contains a capture group so you shouldn't wrap it in one. You could wrap it in a non-capture group `(?:)` to make it optional.

So it'll look like this:

    (?:
        \.?([-_:a-zA-Z0-9]+) # 2: standalone class name
    )?
    [ ]*
    (?:
        '.$this->id_class_attr_catch_re.' # 3: Extra attributes
    )?